pkg/client/results/processing.go

```diff
@@ -52,84 +53,90 @@ type postProcessor func(string, string) (Item, error)
 // file only or all files with a given suffix (for instance).
 type fileSelector func(string, os.FileInfo) bool
 
-// manualResultsAggregation is custom logic just for aggregating results for the top level summary
-// when the plugin is providing the YAML results manually. This is required in (at least) some cases
-// such as daemonsets when each plugin-node will have a result that needs bubbled up to a single,
-// summary Item. This is so that in large clusters you don't have a single plugin have results that
-// scale linearly with the number of nodes and may become unreasonable to show the user.
-//
-// If there is only one top level item, its status is returned. Otherwise a human readable string
-// is produced to show the counts of various values. E.g. "passed: 3, failed: 2, custom msg: 1".
-// Avoiding complete aggregation to avoid forcing a narrow set of use-cases from dominating.
-func manualResultsAggregation(items ...Item) string {
-	// Avoid the situation where we get 0 results (because the plugin partially failed to run)
-	// but we report it as passed.
-	if len(items) == 0 {
-		return StatusUnknown
-	}
-
-	results := map[string]int{}
-	var keys []string
-
+// hasCustomValues returns true if there is a leaf node with a custom status value.
+func hasCustomValues(items ...Item) bool {
```

Comment:
FYI This method simplified the resulting logic. Without it you had to modify the Item structure (or create another and pass it around) to track whether or not custom values were seen. Otherwise two separate branches of the tree (think the first/second node) may differ. E.g. the LHS may only have pass/fail and therefore aggregate to only "pass" or "fail" (a single value) but the RHS would have custom ("pass" and "foo") and therefore count each (e.g. "foo: 1, pass: 2"). Then we've lost the count data for the LHS when we realize we need it.